recipes/libsodium/all/conanfile.py

```diff
@@ -28,54 +32,89 @@ class LibsodiumConan(ConanFile):
         "PIE": False,
     }
 
-    short_paths = True
-
-    _autotools = None
-
     @property
-    def _source_subfolder(self):
-        return "source_subfolder"
+    def _is_mingw(self):
+        return self.settings.os == "Windows" and self.settings.compiler == "gcc"
 
     @property
     def _settings_build(self):
         return getattr(self, "settings_build", self.settings)
 
-    @property
-    def _is_msvc(self):
-        return str(self.settings.compiler) in ["Visual Studio", "msvc"]
-
-    @property
-    def _is_mingw(self):
-        return self.settings.os == "Windows" and self.settings.compiler == "gcc"
-
     def export_sources(self):
-        for patch in self.conan_data.get("patches", {}).get(self.version, []):
-            self.copy(patch["patch_file"])
+        export_conandata_patches(self)
 
     def config_options(self):
         if self.settings.os == "Windows":
             del self.options.fPIC
 
     def configure(self):
         if self.options.shared:
-            del self.options.fPIC
-        del self.settings.compiler.libcxx
-        del self.settings.compiler.cppstd
+            try:
+                del self.options.fPIC
+            except Exception:
+                pass
+        try:
+            del self.settings.compiler.libcxx
+        except Exception:
+            pass
+        try:
+            del self.settings.compiler.cppstd
+        except Exception:
+            pass
+
+    def layout(self):
+        if is_msvc(self):
+            vs_layout(self)
+        else:
+            basic_layout(self, src_folder="src")
 
     def validate(self):
-        if self.options.shared and self._is_msvc and "MT" in msvc_runtime_flag(self):
+        if self.options.shared and is_msvc(self) and "MT" in msvc_runtime_flag(self):
```

Comment:
https://docs.conan.io/en/latest/reference/conanfile/tools/microsoft.html?highlight=is_msvc_static_runtime#conan-tools-microsoft-is-msvc-static-runtime 😉 

```suggestion
        if self.options.shared and is_msvc(self) and is_msvc_static_runtime(self):
```

Reply:
Love it, thanks